File: 프로그래머스/Level2/파일명 정렬.py
# https://programmers.co.kr/learn/courses/30/lessons/17686

def solution(files):
    temp = []
    for file in files:
        head, number, tail = '', '', ''
        check = False
        for i in range(len(file)):
            if file[i].isdigit() == True:
                number += file[i]
                check = True
            elif check == False:
                head += file[i]
            else:
                tail = file[i:]
                break
        temp.append((head, number, tail))
    temp.sort(key=lambda x: (x[0].upper(), int(x[1])))
    return [''.join(t) for t in temp]

File: 프로그래머스/Level2/파일명 정렬.js
// https://programmers.co.kr/learn/courses/30/lessons/17686

const solution = (files) => {
	const pattern = /(\D+)(\d+)/;
	return files.sort((a, b) => {
		const [aHead, aNUM] = a.match(pattern).slice(1, 3);
		const [bHead, bNUM] = b.match(pattern).slice(1, 3);
		if (aHead.toLowerCase() > bHead.toLowerCase()) return 1;
		if (aHead.toLowerCase() < bHead.toLowerCase()) return -1;
		return Number(aNUM) - Number(bNUM);
	});
};
